postgres/spec/feature_matrix.yaml

```diff
@@ -629,6 +634,18 @@ categories:
           The operator provides a mechanism to expose all the logs of the managed Postgres instances to a centralized logging tool.
           The logs must be decorated with extra metadata that includes the Pod name and namespace, the cluster name, the role of the Postgres instance (e.g. primary, replica, standby-leader, etc.) and the timestamp that will be available to be used to filter logs entries.
           There is no need to configure the tool in order to obtain required extra metadata from the logs.
+      - id: stdout
```

Comment:
This feature is only available from 15 onwards, so you'd still need something available for the other supported releases (11, 12, 13, and 14).

Additionally, in our experience we've had to encapsulate the JSON log coming from Postgres in a more standard format, given that JSON log is activated only when the postmaster is started. Other auxiliary processes - or extensions (see pgaudit) - that run alongside the Postgres instance do not in any case support the JSON log format - I am thinking of `initdb` or `pg_ctl` or any backup command (see for example a WAL archiver process). So, in my opinion this is an important capability that is also inline with the "Kubernetes native-ness" of an operator as explained in the inspiring book by Patrick and Jeff.

Regarding the recommended Kubernetes approach, I am referring to the node level log agent - which, in order to work, requires the application to write directly to stdout/stderr and preferably - for optimal routing - in JSON.

Reply:
> This feature is only available from 15 onwards, so you'd still need something available for the other supported releases (11, 12, 13, and 14).

Fair enough. Then I'd improve the wording of the feature to make this very clear: that the feature "is assumed to be provided" for Postgres 15+ and may or may not be provided for earlier versions. You may also add a note to the `vendor_compliance` field to request vendors to clarify in the comments if the functionality is provided only for PG15+ or for a wider version set.

I also believe that after PG14 EOL we may remove the feature --on a future version of the spec-- as all non-EOLed versions would support it.

> Additionally, in our experience we've had to encapsulate the JSON log coming from Postgres in a more standard format [...]

I think this is great functionality, and my opinion is that fits perfectly in the `comments` section of CNPG's vendor submission for this feature.

> Regarding the recommended Kubernetes approach, I am referring to the node level log agent - which, in order to work, requires the application to write directly to stdout/stderr and preferably - for optimal routing - in JSON.

I understand. But what I mean is that considering this a best practice is arguable, as there are other ways to do it and different people may consider them better/worse. For one, there are systems which use a specialized sidecar to collect logs and send them over the network, using more compact forms than JSON, and therefore not requiring JSON. Without getting into which design is better, my comment is to remove or soften the wording about JSON over stderr as the "preferred" way as this may be debatable.

Reply:
Ok. Will use `vendor_compliance` and `comments`